src/Razor/Microsoft.AspNetCore.Razor.Language/src/Legacy/LineTrackingStringBuffer.cs

```diff
@@ -22,17 +21,16 @@ public LineTrackingStringBuffer(string content, string filePath)
 
         public LineTrackingStringBuffer(char[] content, string filePath)
         {
-            _endLine = new TextLine(0, 0);
-            _lines = new List<TextLine>() { _endLine };
+            _lines = new List<TextLine>();
 
-            Append(content);
+            BuildTextLines(content);
 
             _filePath = filePath;
         }
 
         public int Length
         {
-            get { return _endLine.End; }
+            get { return _lines[_lines.Count - 1].End; }
```

Comment:
~How does this not throw for empty files?~. I see line 75 would create an empty TextLine 👍